[MISC][JH] toast a11y enhancements

jhyeep opened the pull request:
**Changes**
- Add ability to dismiss focused toast using `esc` key
- Add `aria-hidden` to icons
- Add "Close notification" `aria-label` for dismiss button
- Add `role=alert` for screenreader accessibility
- Flatten `label/description` structure by removing `div` wrapper (_might_ be breaking change? if using CSS selector)

- delete branch

**Changelog entry**
- A11y enhancements for `Toast` component

**Additional information**
- You may refer to [flagship a11y annotations](https://www.figma.com/design/wDAxbRmlq4xOn4UJ8zPuEF/%F0%9F%A7%91%F0%9F%8F%BB%E2%80%8D%F0%9F%A6%AF-Flagship-A11y-Annotations?node-id=1-28252&p=f&m=dev)

qroll reviewed (changes requested):
on Voiceover/Safari, it is repeating the type and buttons twice, is there any way around this? I'm thinking type could be rendered as a VisuallyHidden element, but button still doesn't work. unless we restrict the alert role to the TextContainer only

<img width="647" height="330" alt="Screenshot 2025-08-05 at 1 48 14 PM" src="https://github.com/user-attachments/assets/bb15c9a6-59b8-4b82-9b0e-2b03e8ad877b" />

qroll commented on src/toast/toast.tsx:
```diff
@@ -98,54 +123,57 @@ export const Toast = ({
         } else if (title) {
-            return <Typography.BodyMD>{label}</Typography.BodyMD>;
+            return (
+                <DescriptionMD $type={type} role="presentation">
```
why does `presentation` role need to be set?

jhyeep replied:
oh it doesn't, forgot to remove it after flattening

jhyeep commented:
> on Voiceover/Safari, it is repeating the type and buttons twice, is there any way around this? I'm thinking type could be rendered as a VisuallyHidden element, but button still doesn't work. unless we restrict the alert role to the TextContainer only
> 
> <img alt="Screenshot 2025-08-05 at 1 48 14 PM" width="647" height="330" src="https://private-user-images.githubusercontent.com/16853062/474335667-bb15c9a6-59b8-4b82-9b0e-2b03e8ad877b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzNzM0ODksIm5iZiI6MTc1NDM3MzE4OSwicGF0aCI6Ii8xNjg1MzA2Mi80NzQzMzU2NjctYmIxNWM5YTYtNTliOC00YjgyLTliMGUtMmIwM2U4YWQ4NzdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA1VDA1NTMwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2MWUwMTJlOTc1MDZhMDFhYTFhNzg3OGM0OGM0YTBjOTFiNzQxMGY0ZjA3YmI5OTc3OGRiMTdmZTBhNzllZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lp8coBzFc6d0HKFeK0Z-w-N1VOcB6pkSk8KR2R5yo0E">

didn't realize VO could somehow have different behavior across browsers, fixed this by avoiding `aria-labels` entirely and using `VisuallyHidden`. also added `role=presentation` for the dismiss `VisuallyHidden` since Chrome was the one giving double readings afterwards LOL

jhyeep commented on src/toast/toast.tsx:
```diff
@@ -98,54 +123,49 @@ export const Toast = ({
+            return title;
+        } else {
+            return (
+                <Title $type={type} weight="semibold" role="presentation">
```
oh this one is intentional to avoid reader announcing "header level _" together with the title, or is that what we want?

qroll replied:
hmm I see, so we are intentionally stripping out the heading semantic. since toast is usually a temporary element and it's not focused it I think that's ok
